adapter/internal/oasparser/envoyconf/routes_with_clusters.go

```diff
@@ -379,6 +390,103 @@ func CreateLuaCluster(interceptorCerts map[string][]byte, endpoint model.Interce
 	return processEndpoints(endpoint.ClusterName, &endpoint.EndpointCluster, interceptorCerts, endpoint.ClusterTimeout, endpoint.EndpointCluster.Endpoints[0].Basepath)
 }
 
+// CreateRateLimitCluster creates cluster relevant to the rate limit service
+func CreateRateLimitCluster() (*clusterv3.Cluster, []*corev3.Address, error) {
+	conf, _ := config.ReadConfigs()
+	var sslCertSanHostName string = ""
+	if conf.Envoy.RateLimit.SSLCertSANHostname == "" {
+		sslCertSanHostName = conf.Envoy.RateLimit.Host
+	} else {
+		sslCertSanHostName = conf.Envoy.RateLimit.SSLCertSANHostname
+	}
+	rlCluster := &model.EndpointCluster{
+		Endpoints: []model.Endpoint{
+			{
+				Host:    conf.Envoy.RateLimit.Host,
+				URLType: httpsURLType,
+				Port:    conf.Envoy.RateLimit.Port,
+			},
+		},
+	}
+	cluster, address, rlErr := processEndpoints(rateLimitClusterName, rlCluster, nil, 20, "")
+	config := &upstreams.HttpProtocolOptions{
+		UpstreamHttpProtocolOptions: &corev3.UpstreamHttpProtocolOptions{
+			AutoSni: true,
+		},
+		UpstreamProtocolOptions: &upstreams.HttpProtocolOptions_ExplicitHttpConfig_{
+			ExplicitHttpConfig: &upstreams.HttpProtocolOptions_ExplicitHttpConfig{
+				ProtocolConfig: &upstreams.HttpProtocolOptions_ExplicitHttpConfig_Http2ProtocolOptions{
+					Http2ProtocolOptions: &corev3.Http2ProtocolOptions{},
+				},
+			},
+		},
+	}
+	MarshalledHTTPProtocolOptions, err := proto.Marshal(config)
+	if err != nil {
+		logger.LoggerOasparser.Errorf("Error occurred while parsing rate-limit HTTP protocol options. Error: %s", err)
```

Comment:
log the error and continue? or should we return the error to the caller without continuing